### DISCUSSION TITLE
PADFramework

### DISCUSSION THREAD

**Discussion by AgniusBartninkas** [AgniusBartninkas/power-automate-desktop-framework#38] [Announcements]
The initial release of fully packaged PADFramework solution, including all components, available for import into Power Platform environments.

<hr /><em>This discussion was created from the release <a href='https://github.com/AgniusBartninkas/power-automate-desktop-framework/releases/tag/v1.0.02407.1937'>PADFramework</a>.</em>

**Comment by NilsOelfke** [comment]
Thank you for uploading this framework.

My question is: What should this framework be used for? Is it to make my work easier, e.g. when I open Excel, I simply copy the Excelflow and can then work with it? Or is there another field of application?
I have looked through the flows a bit and find it really useful, but I would like to understand better what the full potential of the framework is.

**Reply by AgniusBartninkas** [reply]
Thank you for the question.

The idea behind this framework is to help build standardized flows that include the most important components (error handling, logging, reporting, external configurations, etc.,), as well as some modular utility flows that help reduce code repetition.

So, essentially, it helps making your flows standardized, which means it is easier to understand them, provide support and maintenance when needed, even if it's been a while since you last visited them. Also even when support and maintenance is done by people other than those who built the flow (e.g. if the original author of the flow has left the org or is on leave).

Also, another major reason for using it is the fact that most major components are already there, so it significantly reduces the time needed to build a new flow. You do not need to build the structure of your flows, as well as all the error handling, logging and other functionalities, as they're already there. You just need to build the custom bits.

The way to use it has been described on a live session I did recently. See its recording here: https://www.youtube.com/watch?v=taw6_7v6C2s&ab_channel=WhatTheFlow

**Reply by NilsOelfke** [reply]
Ah ok, thx for the explanation and the youtube link :)
